streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -95,7 +98,12 @@ StampedRecord nextRecord(final RecordInfo info) {
                 }
 
                 // always update the stream time to the record's timestamp yet to be processed if it is larger
-                streamTime = Math.max(streamTime, record.timestamp);
+                if (record.timestamp > streamTime) {
+                    streamTime = record.timestamp;
+                    recordLatenessSensor.record(0);
+                } else {
+                    recordLatenessSensor.record(streamTime - record.timestamp);
+                }
```

Comment:
So, to return to @bbejeck 's concern... This metric gets updated once for every record we process. This may be less frequent than for every record in every call to a window aggregator, but it's still pretty darn frequent.

Should I make it DEBUG-level until KAFKA-7275 is done?

Reply:
I'm +1 on keeping both `dropped` and `lateness` sensors DEBUG for now.

Reply:
The `dropped` metric was pre-existing (was `skipped-records` before). It has been at INFO level since 1.1 or 1.2. I think we should leave it at INFO. (The prior justification for this was that it should be abnormal to be dropping records, and it would probably be nice to know by default.)

I have made `lateness` a DEBUG metric, though.